language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -642,7 +639,14 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
             }
 
             // TODO: Handle type actuals for generics
-            Bytecode::Exists(_, _) => {
+            Bytecode::Exists(idx, _) => {
+                let struct_definition = self.module.struct_def_at(*idx);
+                if !StructDefinitionView::new(self.module, struct_definition).is_resource() {
+                    return Err(VMStaticViolation::ExistsNoResourceError(offset));
+                } else if !state.global(*idx).is_empty() {
+                    return Err(VMStaticViolation::GlobalReferenceError(offset));
```

Comment:
Do we need to report an error in this case?

Reply:
Fixed.